### ISSUE TITLE
Saved question of model with long title doesn't work in graphical query

### ISSUE THREAD

**Issue by Dynnammo** [metabase/metabase#35252]
### Context
As a data analyst, I want to create a simple question that groups things from a saved question/model.

### Error
When creating an indicator from a saved question that has a column with a very long title, it doesn't work making a summary on such column

### Expected behavior
Creating

# Debugging info
Seems the column is saved under an alias since the column name is too long.

```sql
select
    id as "ID",
    text as "This is a very very long column title that makes my saved question break when I want to use it elsewhere"
from my_table
```

![image](https://github.com/metabase/metabase/assets/33259633/1a4bd2f1-77be-448c-8f7c-2089badf882f)

When trying to convert the saved question is SQL, I see that the column is aliased
```sql
SELECT
  "source"."col_1" AS "col_1",
  "source"."This is a very very long column title that makes my saved quest" AS "This is a very very long column title that makes my_4090d833"
FROM
  (
    select
      id as "ID",
      text as "This is a very very long column title that makes my saved question break when I want to use it elsewhere"
    from
      mytable
  ) AS "source"
LIMIT
  1048575
```

### To Reproduce

1. Create a saved question with a very long column title
2. Create another question out of it that makes a summary (counting line, sum, etc.)

### Logs

Cannot update binned field: could not find matching source metadata for Field 'This is a very very long column title that makes my_4090d833'

### Information about your Metabase installation

```JSON
- Metabase 0.47.6
- Jar installation
- Original database of the query : Postgres
```

### Severity

Low

**Comment by paoliniluis** [comment]
Can´t reproduce it on master... have you refreshed the question?

**Comment by Dynnammo** [comment]
Yes, I've also cleared the cache in admin panel and disconnect/reconnect, as it sometimes works for refreshing cache of saved questions/models, but both didn't helped.I wonder

**Comment by paoliniluis** [comment]
Aliasing is strictly related to Metabase, but I'm not able to hit the issue unfortunately

**Comment by ignacio-mb** [comment]
I was able to reproduce. To test:
1. Create a SQL question referencing any table and rename one column as "This is a very very long column title that makes my saved question break when I want to use it elsewhere"
2. Create a GUI query out of that saved SQL question and try to Summarize (count, sum, etc) ad group by the long name column.
3. See the error. The error when you un-bin is: ERROR: column source.This is a very very long column title that makes my_4090d833 does not exist Position: 124
4. To see that it works with short column names, go back to the SQL and change the column name to "test". Go back to the GUI query and group by the column "test", the summarization works correctly. 

Version 47.x, postgres

**Comment by ranquild** [comment]
Repro on master `a5c3bec1fad998eaecc3f6fa9fafac6214b35083`

1. New -> SQL -> `select id as "ID", category as "This is a very very long column title that makes my saved question break when I want to use it elsewhere" from products` -> Save as `Q1`
2. New -> Question -> `Q1` -> Filter by the long column name
3. See the error

<img src="https://uploads.linear.app/a24a1a82-ddc8-4567-aacc-c3804ea2385e/fc8b36b6-4fce-444d-ba5e-07d7730c744d/752bea77-7803-424e-8144-ee52933ba3fd?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2EyNGExYTgyLWRkYzgtNDU2Ny1hYWNjLWMzODA0ZWEyMzg1ZS9mYzhiMzZiNi00ZmNlLTQ0NGQtYmE1ZS0wN2Q3NzMwYzc0NGQvNzUyYmVhNzctNzgwMy00MjRlLTgxNDQtZWU1MjkzM2JhM2ZkIiwiaWF0IjoxNzUxNTY0MDUyLCJleHAiOjMzMzIyMTI0MDUyfQ.-o1UzlqZDtBJdDCdf2dbwbQFc4STyu_ZNUUuYgj-Psw " alt="Screenshot 2025-07-03 at 13.32.19.png" width="1234" data-linear-height="609" />

As with join aliases, I think we should not do truncation of aliases in field refs and defer this to QP/drivers.

**Comment by ranquild** [comment]
Can be still reproduced after https://github.com/metabase/metabase/pull/60371
